### PR TITLE
MM-21466 mmctl deleteUsersCmdF Review

### PR DESCRIPTION
**Pull request by IronOnet**
#### Summary

This PR implements a refactor of the deleteUsersCmdF, the aim of the 
refactor improve error handling for this function

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/21466
Jira https://mattermost.atlassian.net/browse/MM-47820

#### Release Note
NONE

### CODE REVIEW

**Comment by IronOnet** [comment]
@mgdelacroix @isacikgoz Let me fix the errors,

**Comment by IronOnet** [comment]
@mgdelacroix @isacikgoz I've fixed the e2e tests

**Comment by isacikgoz** [comment]
@toninis the tests didn't run for this PR, could it be related to closing and re-opening the PR?

**Comment by toninis** [comment]
@isacikgoz Tests are not triggered because the conditions have not met . 
This is a fork PR and the condition is that Server tests are passing . 
In any case we can use [this](https://github.com/mattermost/delivery-platform/tree/master/automations/mattermost#manual-trigger) documentation for manual trigger .

**Comment by IronOnet** [comment]
@mgdelacroix ,  @isacikgoz , I've completed this issue, need a review😊😊

File: server/cmd/mmctl/commands/user.go [hunk 1]
```diff
@@ -638,13 +638,16 @@ func resetUserMfaCmdF(c client.Client, cmd *cobra.Command, args []string) error
...
 		if err := getConfirmation("Are you sure you want to delete the users specified? All data will be permanently deleted?", true); err != nil {
-			return err
+			result = multierror.Append(result, err)
+			return result.ErrorOrNil()
 		}
 	}
 
 	users, err := getUsersFromArgs(c, args)
+	result = multierror.Append(result, err)
 	if err != nil {
```
Comment (isacikgoz) [inline server/cmd/mmctl/commands/user.go:651]:
```suggestion
	if err != nil {
		result = multierror.Append(result, err)
```

**Review by isacikgoz — CHANGES_REQUESTED** [review]
@IronOnet thanks for contribution! I think there is a slight change needed below:

**Comment by IronOnet** [comment]
@isacikgoz i've made the change

**Review by isacikgoz — APPROVED** [review]

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Hey @IronOnet we discussed this offline, I overlooked the remaining part of the code and now I see that this is not a valid use case of the `multierror`. Sorry for the confusion here. 

So, basically we need to use multierror on `L:656` and `L:659`  where we actually receive an error scenario.

**Comment by IronOnet** [comment]
@isacikgoz no probs, I'll take care of it...

**Comment by hanzei** [comment]
@crspeller would you please review the PR as @mgdelacroix is out?

**Pull request closed without merging**